edumanage/models.py

```diff
@@ -427,30 +546,64 @@ class ServiceLoc(models.Model):
     Service Location of an SP/IdPSP Institution
     '''
 
+    GEO_TYPES = get_namedtuple_choices(
+        ('SPOT', 0, 'single spot'),
+        ('AREA', 1, 'area'),
+        ('MOBILE', 2, 'mobile'),
+    )
+
     ENCTYPES = (
         ('WPA/TKIP', 'WPA-TKIP'),
         ('WPA/AES', 'WPA-AES'),
         ('WPA2/TKIP', 'WPA2-TKIP'),
         ('WPA2/AES', 'WPA2-AES'),
     )
 
+    LOCATION_TAGS = (
+        ('port_restrict', 'port restrictions'),
+        ('transp_proxy', 'transparent proxy'),
+        ('IPv6', 'IPv6'),
+        ('NAT', 'NAT'),
+        ('HS2.0', 'Passpoint (Hotspot 2.0)'),
+    )
+
     # accept if institutionid.ertype: 2 (sp) or 3 (idpsp)
     institutionid = models.ForeignKey("Institution", verbose_name="Institution")
-    longitude = models.DecimalField(max_digits=12, decimal_places=8)
-    latitude = models.DecimalField(max_digits=12, decimal_places=8)
+    locationid = models.UUIDField(default=uuid.uuid4, unique=True, editable=False)
+    # single set of coordinates enforced by .signals.sloc_coordinates_enforce_one
+    coordinates = SortedManyToManyField(Coordinates)
+    stage = models.PositiveIntegerField(
+        choices=PRODUCTION_STATES,
+        default=PRODUCTION_STATES.ACTIVE
+    )
+    geo_type = models.PositiveIntegerField(
+        choices=GEO_TYPES,
+        default=GEO_TYPES.SPOT,
+        db_column='type'
+    )
     # TODO: multiple names can be specified [...] name in English is required
     loc_name = fields.GenericRelation(Name_i18n)
-    address_street = models.CharField(max_length=96)
-    address_city = models.CharField(max_length=64)
+    # TODO: multiple addresses can be specified [...] address in English is required
+    address = fields.GenericRelation(Address_i18n)
+    venue_info = models.CharField(
```

Comment:
Same comment as for InstitutionDetails.venue_info:

> Why is the field called`venue_info` but stored in db_column `inst_type` ?  That just looks like unnecessary complexity - can't we just use one name for both?
> 
> I expect there is a reasonable answer to this - just asking because it struck me as odd.

Reply:
See previous

Reply:
Agreed.